pkg/frontend/transport/handler_test.go

```diff
@@ -359,6 +436,9 @@ func TestHandler_ServeHTTP(t *testing.T) {
 			require.NoError(t, err)
 			require.Empty(t, activities)
 
+			if tt.assertHeaders != nil {
+				tt.assertHeaders(t, resp.Header())
```

Comment:
Related to the [discussion above](https://github.com/grafana/mimir/pull/10897/files#r2002657907), let's fix the existing test. Otherwise, we aren't testing what we wanted to.

```suggestion
				tt.assertHeaders(t, resp.Result().Header())
```